src/Compilers/Core/Portable/Diagnostic/ReportDiagnosticExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    public static class ReportDiagnosticExtensions
+    {
+        public static DiagnosticSeverity ToDiagnosticSeverity(this ReportDiagnostic reportDiagnostic, DiagnosticSeverity defaultSeverity)
+        {
+            switch (reportDiagnostic)
+            {
+            case ReportDiagnostic.Error:
+                return DiagnosticSeverity.Error;
+
+            case ReportDiagnostic.Warn:
+                return DiagnosticSeverity.Warning;
+
+            case ReportDiagnostic.Info:
+                return DiagnosticSeverity.Info;
+
+            case ReportDiagnostic.Hidden:
+                return DiagnosticSeverity.Hidden;
+
+            case ReportDiagnostic.Suppress:
```

Comment:
>case ReportDiagnostic.Suppress: [](start = 12, length = 31)

This seems a bit weird. Why are we mapping `ReportDiagnostic.Suppress` to default severity? Should this API return a nullable value and this option map return null?

Reply:
I wasn't particularly happy about this since it wasn't as clear at the call sites as I'd hope. I ended up mapping *both* `Suppress` and `Hidden` to null in 42170541ade79911921043fb002f9455c451c35d, and I believe the change was a nice clarity improvement. I added documentation for the new APIs in 327769804788363dfcac031a24aae9a4df3ec21b.

Reply:
Is it possible to even move this API to the internal DiagnosticHelper type you added in IDE layer? Semantics are still not intuitive for input `ReportDiagnostic.Default` and if we eventually move all our IDE notification options to be just compilation options, only the compiler layer will ever need to do this mapping, and the API would not need to be public anymore.

---
In reply to: [193056255](https://github.com/dotnet/roslyn/pull/27220#discussion_r193056255) [](ancestors = 193056255)